packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/CookieManagerHostApiImpl.java

```diff
@@ -0,0 +1,27 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.webviewflutter;
+
+import android.os.Build;
+import android.webkit.CookieManager;
+
+class CookieManagerHostApiImpl implements GeneratedAndroidWebView.CookieManagerHostApi {
+  @Override
+  public void clearCookies(GeneratedAndroidWebView.Result<Boolean> result) {
+    CookieManager cookieManager = CookieManager.getInstance();
+    final boolean hasCookies = cookieManager.hasCookies();
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      cookieManager.removeAllCookies(value -> result.success(hasCookies));
```

Comment:
This should probably use `value` instead of `hasCookies`. `hasCookies` only exists for versions below lollipop, as `removeAllCookie` does not return a boolean.